cmd/manager/cel-scanner.go

```diff
@@ -0,0 +1,680 @@
+	if celConf.ScanType == "Platform" {
+		scanner.runPlatformScan()
+	} else {
+		cmdLog.Error(nil, "Unsupported scan type", "scanType", celConf.ScanType)
+		os.Exit(CelExitCodeError)
+	}
+}
+
+// runPlatformScan runs the platform scan based on the profile and inputs.
+func (c *CelScanner) runPlatformScan() {
+	cmdLog.V(1).Info("Running platform scan")
+	// Load and parse the profile
+	profile := c.celConfig.Profile
+	if profile == "" {
+		cmdLog.Error(nil, "Profile not provided", "scanName", c.celConfig.ScanName)
+		os.Exit(CelExitCodeError)
+	}
+	exitCode := CelExitCodeCompliant
+	// Check if a tailored profile is provided, and get selected rules
+	var selectedRules []*cmpv1alpha1.CustomRule
+	var setVars []*cmpv1alpha1.Variable
+	if c.celConfig.Tailoring != "" {
+		tailoredProfile, err := c.getTailoredProfile(c.celConfig.NameSpace)
+		if err != nil {
+			cmdLog.Error(err, "Failed to get tailored profile", "name", c.celConfig.Tailoring)
+			os.Exit(CelExitCodeError)
+		}
+		selectedRules, err = c.getSelectedCustomRules(tailoredProfile)
+		if err != nil {
+			cmdLog.Error(err, "Failed to get selected rules for tailored profile", "name", c.celConfig.Tailoring)
+			os.Exit(CelExitCodeError)
+		}
+		// Collect all the variables being set in the tailoredProfile
+		setVars, err = c.getVariablesForTailoredProfile(tailoredProfile)
+		if err != nil {
+			cmdLog.Error(err, "Failed to get set variables for tailored profile", "name", c.celConfig.Tailoring)
+			os.Exit(CelExitCodeError)
+		}
+	} else {
+		cmdLog.Error(nil, "No tailored profile provided")
+		os.Exit(CelExitCodeError)
+	}
+
+	// Convert variables to SDK format
+	celVariables := make([]scanner.CelVariable, 0, len(setVars))
+	for _, v := range setVars {
+		// Create an inline adapter for Variable to CelVariable
+		celVar := &celVariableAdapter{
+			name:      v.Name,
+			namespace: v.Namespace,
+			value:     v.Value,
+		}
+		celVariables = append(celVariables, celVar)
+	}
+
+	// Convert CustomRules to SDK Rules
+	// CustomRule now properly implements scanner.Rule and scanner.CelRule interfaces
+	sdkRules := make([]scanner.Rule, 0, len(selectedRules))
+	for _, customRule := range selectedRules {
+		if customRule.Spec.CustomRulePayload.Expression == "" {
+			cmdLog.Info("Warning: Skipping rule with empty expression", "rule", customRule.Name)
+			continue
+		}
+		sdkRules = append(sdkRules, customRule)
+	}
+
+	// Create scan configuration
+	// Note: ApiResourcePath in the SDK expects the cache directory path
+	scanConfig := scanner.ScanConfig{
+		Rules:              sdkRules,
+		Variables:          celVariables,
+		ApiResourcePath:    c.celConfig.ApiResourceCacheDir,
+		EnableDebugLogging: debugLog,
+	}
+
+	// Run the scan using SDK scanner
+	ctx := context.Background()
+	checkResults, err := c.sdkScanner.Scan(ctx, scanConfig)
+	if err != nil {
+		cmdLog.Error(err, "Failed to run scan", "scanName", c.celConfig.ScanName, "profile", c.celConfig.Profile)
```

Comment:
Are we scanning with `profile` or with `tailoredProfile`?

Initially we will only support `tailoredProfile`, right?